File: leetcode/rust-leetcode/src/1605.find-valid-matrix-given-row-and-column-sums.rs
use std::cmp::min;
impl Solution {
    pub fn restore_matrix(mut row_sum: Vec<i32>, mut col_sum: Vec<i32>) -> Vec<Vec<i32>> {
        let (m, n) = (row_sum.len(), col_sum.len());
        let mut matrix = vec![vec![0; n]; m];
        let mut x = 0;
        for i in 0..m {
            for j in 0..n {
                x = min(row_sum[i], col_sum[j]);
                matrix[i][j] = x;
                row_sum[i] -= x;
                col_sum[j] -= x;
            }
        }
        matrix
    }
}

File: leetcode/go/1605.find-valid-matrix-given-row-and-column-sums.go
func restoreMatrix(rowSum []int, colSum []int) [][]int {
    m, n, cell := len(rowSum), len(colSum), 0
    mat := make([][]int, m)
    for i := 0; i < m ; i++ {
        mat[i] = make([]int, n)
        for j := 0; j < n ; j++ {
            cell = min(rowSum[i], colSum[j])
            mat[i][j] = cell
            rowSum[i] -= cell
            colSum[j] -= cell
        }
    }
    return mat
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

